public/app/plugins/datasource/prometheus/language_utils.ts

```diff
@@ -111,3 +113,34 @@ export function expandRecordingRules(query: string, mapping: { [name: string]: s
   const rulesRegex = new RegExp(`(\\s|^)(${ruleNames.join('|')})(\\s|$|\\(|\\[|\\{)`, 'ig');
   return query.replace(rulesRegex, (match, pre, name, post) => `${pre}${mapping[name]}${post}`);
 }
+
+/**
+ * Adds metadata for synthetic summary metrics for which the API does not provide metadata.
+ * See https://github.com/grafana/grafana/issues/22337 for details.
+ *
+ * @param metadata HELP and TYPE metadata from /api/v1/metadata
+ */
+export function fixSummariesMetadata(metadata: PromMetricsMetadata): PromMetricsMetadata {
+  if (!metadata) {
+    return metadata;
+  }
+  const summaryMetadata: PromMetricsMetadata = {};
+  for (const metric in metadata) {
+    const item = metadata[metric][0];
+    if (item.type === 'summary') {
```

Comment:
Do we want to add the other cases as part of this PR as well? I think we should be taking `_bucket`, `_sum`, `_count`, and `_total` into account as well.

In principle, I think I like modifying the metadata for these metrics with a clear explanation of what they actually are. However, it's one more level of indirection for people trying to look at what the base metrics does/is. 

To be clear, I'm not opposed to this I just think it's worth a discussion.

Reply:
Yes, lets do histograms as well. They seem so consist of `_bucket`, `_sum`, and `_count`. 

Can you explain what mechanism exists around `_total`?

Reply:
How do I find the metadata for the histogram base metric? I don't think `foo` is part of the metadata API result if a metric `foo_bucket` exists.

Reply:
> Can you explain what mechanism exists around _total?

My bad, we don't need to do anything for `_total` only the other three. 

> How do I find the metadata for the histogram base metric? I don't think foo is part of the metadata API result if a metric foo_bucket exists.

It's already there if I understand the question correctly. The metadata API won't expose the `_bucket` metric, the only thing that'll show up is the `foo` metric with the type of histogram.